content/docs/user-guide/data-pipelines/defining-pipelines.md

```diff
@@ -0,0 +1,237 @@
+# Defining Pipelines
+
+DVC pipelines are written as collections of data processing <abbr>stages</abbr>
+in `dvc.yaml` files. These represent data workflows that you want to
+**reproduce** reliably later. Here's a sample 3-stage structure:
+
+```yaml
+stages:
+  prepare: ... # stage 1 definition
+  train: ... # stage 2 definition
+  evaluate: ... # stage 3 definition
+```
+
+<admon>
+
+This _codification_ has the added benefit of allowing you to develop pipelines
+on standard Git workflows (GitOps).
+
+</admon>
+
+A pipeline is formed by making [stages](#stages) interdependent, meaning that
+the output of one becomes the input of another, and so on. Technically, this is
+called a _dependency graph_ (DAG).
+
+## Directed Acyclic Graph (DAG)
+
+DVC represents a pipeline internally as a _graph_ where the nodes are stages and
+the edges are _directed_ dependencies (e.g. A before B). In order for DVC to
+execute the pipeline reliably, its topology should be _acyclic_ -- because
+executing cycles (e.g. A -> B -> C -> A ...) would continue indefinitely. [More
+about DAGs].
+
+<admon type="info">
+
+Stage execution will be determined entirely by the DAG, not by the order in
+which stages are found in `dvc.yaml`.
+
+</admon>
+
+Use `dvc dag` to visualize (or export) pipeline DAGs.
+
+[more about dags]: https://en.wikipedia.org/wiki/Directed_acyclic_graph
+
+## Stages
+
+<admon type="tip">
+
+See the full [specification] of stage entries.
+
+[specification]: /doc/user-guide/project-structure/dvcyaml-files#stage-entries
+
+</admon>
+
+Each stage wraps around an executable shell [command] and specifies the
```

Comment:
should we mention here or somewhere that the regular process is:

# add DVC tracked inputs
dvc add <some data>
dvc add <some data>

# build pipeline
dvc stage ...

# then metrics, etc?

Reply:
Good idea. I had to rewrite the first section of the doc for this but I think it's better now.